mantis-runtime/src/main/java/io/mantisrx/runtime/parameter/Parameters.java

```diff
@@ -38,22 +37,34 @@ public Parameters(Map<String, Object> state,
         this.parameterDefinitions = parameterDefinitions;
     }
 
+    /**
+     * Get parameter value given key with validation.
+     *
+     * If the key is required, parameters must have a value provided otherwise it will throw an exception.
+     * If the key is not defined, it will throw an exception.
+     */
     public Object get(String key) {
-        // check if required parameter, make sure
-        // has a value
+        // check if required parameter, make sure has a value
         if (requiredParameters.contains(key) &&
                 !state.containsKey(key)) {
-            throw new ParameterException("Attempting to reference a required parameter witn no value: " + key + ", check parameter definitions for job.");
+            throw new ParameterException("Attempting to reference a required parameter witn no value: " + key +
+                ", check parameter definitions for job.");
         }
         // check if parameter definition exists
         if (!parameterDefinitions.contains(key)) {
-            throw new ParameterException("Attempting to reference parameter: " + key + ", with no definition, check parameter definitions in job.");
+            throw new ParameterException("Attempting to reference parameter: " + key +
+                ", with no definition, check parameter definitions in job.");
         }
         return state.get(key);
     }
 
+    /**
+     * Get parameter value given key without validation.
+     *
+     * If the key is not defined or missing a provided value, the given default value will be returned.
+     */
     public Object get(String key, Object defaultValue) {
```

Comment:
What if you have a default value of null for the parameter and pass in a non-default value here as part of defaultValue? What should be the expected behavior? Can we also write tests for this to ensure the behavior remains consistent as we advance?

Reply:
That's a good question, but I really do not have the answer. The expected behavior depends on how it's now being used.

Reply:
According to this code snippet: https://github.com/Netflix/mantis/blob/master/mantis-runtime/src/main/java/io/mantisrx/runtime/parameter/ParameterUtils.java#L309-L315

It seems "default value of null for the parameter" means the parameter does not have a default value, i.e. missing in the definition. In that case, this `get()` method is doing the right thing: return the provided `defaultValue`. We can add a test for this.